tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3877,3 +3875,416 @@ def test_complex128():
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_float32():
+        from numpy import imag, shape, array, float32
+        arr = array([4,5,6,2,1], float32)
+        a = imag(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_float64():
+        from numpy import imag, shape, array, float64
+        arr = array([4,5,6,2,1], float64)
+        a = imag(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_complex64():
+        from numpy import imag, shape, array, complex64
+        arr = array([4,5,6,2,1], complex64)
+        a = imag(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_complex128():
+        from numpy import imag, shape, array, complex128
+        arr = array([4,5,6,2,1], complex128)
+        a = imag(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    f_bl = epyccel(test_bool, language=language)
+    assert (f_bl() == test_bool())
+
+    f_integer = epyccel(test_int, language=language)
+
+    # should be uncommented after resolving #733
+    # int8 and int16 numpy data types not recognised by Pyccel.
+    #f_integer8 = epyccel(test_int8, language=language)
+    #f_integer16 = epyccel(test_int16, language=language)
+    f_integer32 = epyccel(test_int32, language=language)
+    f_integer64 = epyccel(test_int64, language=language)
+
+    assert (f_integer() == test_int())
+
+    # should be uncommented after resolving #733
+    # int8 and int16 numpy data types not recognised by Pyccel.
+    #assert (f_integer8() == test_int8())
+    #assert (f_integer16() == test_int16())
+    assert (f_integer32() == test_int32())
+    assert (f_integer64() == test_int64())
+
+    f_fl = epyccel(test_float, language=language)
+    f_fl32 = epyccel(test_float32, language=language)
+    f_fl64 = epyccel(test_float64, language=language)
+
+    assert (f_fl() == test_float())
+    assert (f_fl32() == test_float32())
+    assert (f_fl64() == test_float64())
+
+    f_complex64 = epyccel(test_complex64, language=language)
+    f_complex128 = epyccel(test_complex128, language=language)
+
+    assert (f_complex64() == test_complex64())
+    assert (f_complex128() == test_complex128())
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Tuples not implemented yet"),
+            pytest.mark.c]
+        )
+    )
+)
+
+def test_numpy_imag_array_like_2d(language):
+
+    def test_bool():
+        from numpy import imag, shape, array
+        arr = array([[4,5,6,2,1],[4,5,6,2,1]], bool)
+        a = imag(arr)
+        s = shape(a)
+        return len(s), s[0], s[1], a[0,1], a[1,0]
+
+    def test_int():
+        from numpy import imag, shape, array, int as NumpyInt
+        arr = array([[4,5,6,2,1],[4,5,6,2,1]], NumpyInt)
+        a = imag(arr)
+        s = shape(a)
+        return len(s), s[0], s[1], a[0,1], a[1,0]
```

Comment:
for `test_numpy_imag_array_like_2d` can you use random values with the suggested types, and I think that it will be better to separate tests with functions, each function handle a case, so when the test is failed it will be easy to recognized and deal with it.